### DISCUSSION TITLE
Information in the User model

### DISCUSSION THREAD

**Discussion by Samy-33** [BitByte-TPC/gymkhana#34] [Q&A] [answered]
Hi folks,

This discussion is opened to get everyone's opinion about what information about the user we need to store. I hope everyone thinks a little bit about it and recommends what they think.

Feel free to suggest something even when you think it might not fit. But do share your thought process behind particular fields we will store.

Saket

**Comment by garg3133** [comment, marked as answer]
**Common:**
* First name, Last name
* Email id (to be used as username, first point of contact)
* Contact no. (for emergency contacts, will be visible to office bearers of clubs only)
* Display Picture (optional)
* User type (student/faculty/staff)

**Student:**
* Roll no.
* Batch
* Department
* Hostel address (only visible to office bearers, but not sure about this)
* Social profile links (Linkedin, Github, etc. as per club's preferences; presence of a profile link can be mandated by a club during registration)
* Bio (if they wish to write something about themselves which will appear on their public [inside the institute] profile).

**Faculty:**
* Title (Prof./Dr.)
* Department
* Designation (like Assistant Professor, not a necessity to keep this but won't harm)

**Staff:**
* Department (like DSA Office)
* Designation (like Registrar)

Will update this if I think of any other information we should/need to store.

**Reply by Samy-33** [reply]
This sounds good for now. We will create 4 models:
- `User` model which will extend django's default AbstractUser model.
- `Student`, `Faculty` and `Staff` models which will have {0,1}:1 mapping with User

We would like to add another field into `User` model: `user_type` which will differentiate between these roles.

**Comment by PranshuNayak** [comment]
I have designed the mock up designs for registration pages . Please take a look and suggest if anything can be improved.
<img width="1440" alt="res_1" src="https://user-images.githubusercontent.com/86283785/173337793-1c311c6f-f113-4c39-8787-7ecc19866f97.png">
<img width="1440" alt="res_2" src="https://user-images.githubusercontent.com/86283785/173337811-6df8770e-b44f-4ac0-b578-29cc654b5815.png">
<img width="1440" alt="res_2" src="https://user-images.githubusercontent.com/86283785/173337825-95c75def-42a9-4db9-90af-60bdddb3449e.png">
<img width="1440" alt="res_3" src="https://user-images.githubusercontent.com/86283785/173337812-89f58929-98fe-40af-89bc-4ba3ca818d9a.png">
